torchci/lib/drciUtils.ts

```diff
@@ -430,23 +430,26 @@ export function isExcludedFromFlakiness(job: RecentWorkflowsData): boolean {
   return isExcluded(job, EXCLUDED_FROM_FLAKINESS);
 }
 
-export async function fetchIssueLabels(
-  octokit: Octokit,
+export async function fetchPRLabels(
```

Comment:
I think you should move the fetchPRLabels query here and the code to clean up failure context to a different PR and keep this one only for GH api caching.  The first twos are BE items, while the latter seems like something we could explore further